setup/setup_snyk.sh

```diff
@@ -49,48 +53,64 @@ esac
 } > snyk
 
 if ! command -v "$SUDO_CMD" &> /dev/null; then
-  echo "$SUDO_CMD is NOT installed. Trying without sudo, expecting privileges to write to '/usr/local/bin'."
+  echo_with_timestamp "$SUDO_CMD is NOT installed. Trying without sudo, expecting privileges to write to '/usr/local/bin'."
   SUDO_CMD=""
 else
-    echo "$SUDO_CMD is installed."
+    echo_with_timestamp "$SUDO_CMD is installed."
 fi
 
 chmod +x snyk
 ${SUDO_CMD} mv snyk /usr/local/bin
 # Function to download a file with fallback to backup URL
 # Parameters:
-#   $1: File name to download
+#   $1: Download URL
 #   $2: Output file name
 download_file() {
-    # Try to download from the main URL
-    if curl --compressed --retry 2 --output "$2" "$MAIN_URL/$1?utm_source="$GH_ACTIONS; then
-        echo "Downloaded from $MAIN_URL/$1"
-    # If main URL fails, try the backup URL
-    elif curl --compressed --retry 2 --output "$2" "$BACKUP_URL/$1?utm_source="$GH_ACTIONS; then
-        echo "Downloaded from $BACKUP_URL/$1"
-    # If both URLs fail, return an error
+    echo_with_timestamp "Downloading Snyk binary"
```

Comment:
nitpick: add download url already here, to know what is going to happen

```suggestion
    echo_with_timestamp "Downloading Snyk binary from $1"
```